### PR TITLE
#162163131 Users should be able to Login to Authors Haven

### PR DESCRIPTION
**Pull request by lindseyme**
### What does this PR do?
This Pull Request enables users to Login to Authors Haven.

#### Description of Task to be completed?
When a user logins to Authors Haven, he/she should be able to access protected routes that require authorization to be used. 

#### How should this be manually tested?
- Clone the repository then checkout to the branch `ft-login-162163131`.
- Install the required dependencies.
     - yarn 
- Start the server.
     - yarn start
- Run the project in the browser.
- Click on the login button on the interface and fill in the details. 

#### What are the relevant pivotal tracker stories?
[#162163131](https://www.pivotaltracker.com/story/show/162163131)

### CODE REVIEW

**Review by sekayasin — COMMENTED** [review]
Hey @lindseyme Bravo,  Let's raise the coverage

**Review by mulondo — APPROVED** [review]
good job

**Review by ongebo — APPROVED** [review]
LGTM

**Review by alexkayabula — APPROVED** [review]
Looks good

File: src/components/auth/Login.js [hunk 1]
```diff
@@ -1,8 +1,60 @@
...
+        this.setState({emailError:nextProps.errors.email})
+      }
+      else if(nextProps.errors.password){
+        this.setState({passwordError:nextProps.errors.password})
+      }
+      else{
+        this.setState({notFoundUser:nextProps.errors.error})
+      }
+      console.log('NEXT PROPS',nextProps.errors)
+      // this.setState({ errors: nextProps.errors.errors });
```
Comment (barema4) [inline src/components/auth/Login.js]: Please remove commented out code

Comment (RrNn) [reply src/components/auth/Login.js]: And the `console.log` too

Comment (lindseyme) [reply src/components/auth/Login.js]: I have worked on that. Thank you for the feedback

File: src/__tests__/components/auth/Login.test.js [hunk 2]
```diff
@@ -0,0 +1,59 @@
...
+    expect(shallow(<Login />).find('#password').length).toEqual(0);
+  });
+
+  const props = {
+    loginUser: (user, history) => jest.fn(),
+    auth: { isAuthenticated: false, user: {} },
+    errors: {}
+  };
+
+  it('test Login component onchange', () => {
```
Comment (RrNn) [inline src/__tests__/components/auth/Login.test.js:42]: This test has no expectations.

File: src/reducers/AuthReducer.js [hunk 3]
```diff
@@ -0,0 +1,12 @@
...
+    user: {}
+};
+export default function(state = initialState, action){
+    switch (action.type) {
+        
+        default:
+        return state;
+
+    }
+}
```
Comment (RrNn) [inline src/reducers/AuthReducer.js:12]: Add a new blank line here

File: src/reducers/ErrorReducer.js [hunk 4]
```diff
@@ -0,0 +1,11 @@
...
+const initialState = {};
+export default function(state = initialState, action){
+    switch (action.type) {
+        case GET_ERRORS:
+            return action.payload;
+        default:
+        return state;
+
+    }
+}
```
Comment (RrNn) [inline src/reducers/ErrorReducer.js:11]: Add a new blank line here

**Review by RrNn — CHANGES_REQUESTED** [review]
These need changing

File: src/reducers/ErrorReducer.js (hunk 4 above)
Comment (lindseyme) [reply src/reducers/ErrorReducer.js:11]: I have worked on that.

**Pull request merged**